src/scenes/Activities/scenes/EventHistory/index.js

```diff
@@ -87,22 +80,11 @@ export default class EventHistory extends Component {
   };
 
   componentWillMount() {
-    const { setAppLocation } = this.props;
-
-    setAppLocation(AppLocation.activityHistory);
     this.executeTxsRequest();
   }
 
   componentWillReceiveProps(nextProps) {
-    const { oracles, transactions, syncBlockNum } = this.props;
-
-    // return from the click event
-    if (nextProps.oracles !== oracles) {
-      const path = getDetailPagePath(nextProps.oracles.data);
```

Comment:
as discussed in the meeting, this logic is here so the correct Oracle detail page is always routed to. 

eg. say an event has 3 voting rounds, if you click on the event from the history table say from voting round 1, you don't want to route to that one cause its not actionable. you should be routed to the latest Oracle detail page (voting round 3).

Reply:
will take a look

Reply:
I made a fix. Eventually once we have mobx, we can just use an `async` function and wait on the api call to complete before we navigate to the next page. We should get this out of the `componentDidUpdate` eventually